### ISSUE TITLE
Bytes in a tantivy document should be encoded in base64 when using schema.to_json(document)

### ISSUE THREAD

**Issue by fmassot** [quickwit-inc/tantivy#1576]
## Problem

When building a document from a json object (method `json_object_to_doc`), for a bytes field, we assume that the bytes are encoded in base64 which is fine!

But when we call `schema.to_json(&document)` we are using `serde_json::to_string` which will not serialize bytes in base64.

Here is a small test showing what's happening.

```rust

    #[test]
    fn test_parse_bytes_doc() {
        let mut schema_builder = Schema::builder();
        let bytes_options = BytesOptions::default();
        let bytes_field = schema_builder.add_bytes_field("my_bytes", bytes_options);
        let schema = schema_builder.build();
        let mut doc = Document::default();
        doc.add_bytes(bytes_field, "this is a test".as_bytes());
        let json_string = schema.to_json(&doc);
        assert_eq!(
            json_string,
            "{my_bytes:\"dGhpcyBpcyBhIHRlc3Q=\"}"
        );
    }
```

## Solution

When serializing a doc to json, bytes should be encoded. in base64 to remain consistent with deserialization.

**Comment by nigel-andrews** [comment]
I don't have the rights to push on this repo so I cannot propose my solution to this issue.

**Comment by adamreichold** [comment]
> I don't have the rights to push on this repo so I cannot propose my solution to this issue.

Why can't you fork this repository and propose a PR from the fork?

**Comment by fmassot** [comment]
Fixed by #1608
